docs/languages/javascript.md

```diff
@@ -0,0 +1,128 @@
+---
+slug: /languages/javascript
+title: JavaScript
+hide_title: true
+description: >-
+  Detailed documentation for Semgrep's JavaScript support. 
+tags:
+  - Semgrep Code 
+---
+
+# JavaScript frameworks and analyses
+
+:::tip 
+Semgrep’s Python coverage leverages framework-specific analysis capabilities that are not present in OSS. As a result, many framework specific Pro rules will **fail** to return findings if run on OSS. To ensure full security coverage, run: `semgrep login && semgrep ci`.
+:::
+
+## Semgrep Code analyses
+
+- Framework-specific control flow analysis
+- Inter-file analysis (cross-file)
+- Inter-procedural analysis (cross-function)
+
+### Coverage
+
+Semgrep aims to provide comprehensive and accurate detection of common OWASP Top 10 issues in source code.
+
+In addition to rules, the Semgrep engine itself can analyze code and implicit dataflows in the context of the following supported frameworks:
+
+
+| Supported frameworks | Type of framework |
+| -------              | ------            |
+| Express              | Web framework     |
+| Koa                  | Web framework     |
+| Hapi                 | Web framework     |
+| NestJS               | Web framework     |
+
+<details>
+<summary>**In addition, Semgrep Code supports 50+ libraries & frameworks based on their overall popularity.**</summary>
```

Comment:
```suggestion
<summary>**Semgrep Code supports 50+ libraries & frameworks based on their overall popularity.**</summary>
```

Removing since we start the previous paragraph with "In addition..."